platform/platform.go

```diff
@@ -326,16 +327,25 @@ func CheckMachine(ctx context.Context, m Machine) error {
 		}
 		out, stderr, err := m.SSH("systemctl is-system-running")
 		if !bytes.Contains([]byte("initializing starting running stopping"), out) {
-			return nil // stop retrying if the system went haywire
+			return fmt.Errorf("machine reached a bad state: %s, %v, %s", out, err, stderr)
```

Comment:
We'll need to build a mechanism for detecting if we're running `kola spawn`. As this code currently is we cannot spawn into machines in a degraded state any longer with this patch.